### PR TITLE
Resolve issue : Somewhat confusing error with extended_key_value_attributes

### LINKED ISSUE #85066: Somewhat confusing error with `extended_key_value_attributes`
### Code

Note that `../README.md` must **not** exist in order to reproduce this error:

```rust
#![feature(extended_key_value_attributes)]
#![doc = include_str!("../README.md")]
```

[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=7ead434499cf146866df4672d2d19ef4)

### Output

The first error is what I would expect, but the second is a bit confusing and weird:

```
error: couldn't read src/../README.md: No such file or directory (os error 2)
 --> src/lib.rs:2:10
  |
2 | #![doc = include_str!("../README.md")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected token: `(/*ERROR*/)`
 --> src/lib.rs:2:10
  |
2 | #![doc = include_str!("../README.md")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors
```

Perhaps instead of emitting ``unexpected token: `(/*ERROR*/)` `` when an error token is encountered, we could use a `delay_span_bug` since an error should have already occurred.

If for some reason an error had *not* already been emitted, the compiler would crash; otherwise, it would just show the original error.

_Originally posted by @camelid in https://github.com/rust-lang/rust/issues/83366#issuecomment-834750746_

### PR DESCRIPTION
**Pull request by Wardenfar**
Fixes #85066

### CODE REVIEW

File: compiler/rustc_parse/src/parser/expr.rs [hunk 1]
```diff
@@ -1493,6 +1493,20 @@ impl<'a> Parser<'a> {
...
+                let expr = match inner.as_ref() {
+                    token::NtExpr(expr) => Some(expr),
+                    token::NtLiteral(expr) => Some(expr),
+                    _ => None,
+                };
+                if let Some(expr) = expr {
+                    if matches!(expr.kind, ExprKind::Err) {
+                        self.diagnostic()
+                            .delay_span_bug(self.token.span, &"invalid interpolated expression");
+                        return self.struct_span_allow(self.token.span, &"ignored message");
```
Comment (nagisa) [inline compiler/rustc_parse/src/parser/expr.rs:1506]: I think you can use `struct_dummy` here, instead.

Comment (Wardenfar) [reply compiler/rustc_parse/src/parser/expr.rs]: i submitted the fix

**Comment by nagisa** [comment]
Please squash the commits in order to retain a clean history after the merge happens.

**Comment by camelid** [comment]
@Wardenfar If you edit the PR description to be `Fixes #85066.`, GitHub will automatically close the issue when this PR is merged.

**Comment by Wardenfar** [comment]
@nagisa I have tried to sqash tell me if it is right
@camelid i have changed the description of the PR

**Comment by JohnTitor** [comment]
Failed in rollup: https://github.com/rust-lang/rust/pull/89087#issuecomment-922433249
@bors r-

**Comment by Wardenfar** [comment]
@JohnTitor i think i fixed it

**Comment by the8472** [comment]
This PR could use a better title

**Pull request merged**